forcedotcom/sfdx-core
# feat: support stronger ciphers

### What does this PR do?
Supports stronger ciphers for encryption/decryption.

### What issues does this PR fix or reference?
@W-12422652@

QA Notes
**MAKE A BACKUP OF YOUR ~/.sfdx DIRECTORY BEFORE DOING ANYTHING.  Use the backup to revert everything done during testing.**

Link this library to plugin-auth and plugin-org.
Enable Crypto debug output.  `export DEBUG=sf:crypto` and `export SF_LOG_LEVEL=trace`
You'll probably want to set `SF_USE_GENERIC_UNIX_KEYCHAIN=true` for initial testing.
Run commands from those plugins without SF_CRYPTO_V2 env var.
Run commands from those plugins with SF_CRYPTO_V2=false.
Run commands from those plugins with SF_CRYPTO_V2=true.
Move the ~/.sfdx dir so that a new key is generated, then:
Run commands from those plugins with SF_CRYPTO_V2=true.

See the doc in the WI for more usecases.
